### ISSUE TITLE
Failed to load BLIP

### ISSUE THREAD

**Issue by alienfractal** [pharmapsychotic/clip-interrogator#33]
Made a clean install and tried the sample snippet on the wiki and kept getting the same error. 
Any advice to get past this point?
Win11
Cuda 11.6
python 3.10.6

```
Loading BLIP model...
Traceback (most recent call last):
  File "C:\ken\style-transfer\diff\clip-interrogator\testblip.py", line 4, in <module>
    ci = Interrogator(Config(clip_model_name="ViT-L-14/openai"))
  File "C:\ken\style-transfer\diff\clip-interrogator\clip_interrogator\clip_interrogator.py", line 58, in __init__
    blip_model = blip_decoder(
  File "c:\ken\style-transfer\diff\clip-interrogator\src\blip\models\blip.py", line 175, in blip_decoder
    model,msg = load_checkpoint(model,pretrained)
  File "c:\ken\style-transfer\diff\clip-interrogator\src\blip\models\blip.py", line 218, in load_checkpoint
    checkpoint = torch.load(cached_file, map_location='cpu')
  File "C:\Users\ken\Anaconda2\envs\StableDiffusion\lib\site-packages\torch\serialization.py", line 777, in load
    with _open_zipfile_reader(opened_file) as opened_zipfile:
  File "C:\Users\ken\Anaconda2\envs\StableDiffusion\lib\site-packages\torch\serialization.py", line 282, in __init__
    super(_open_zipfile_reader, self).__init__(torch._C.PyTorchFileReader(name_or_buffer))
RuntimeError: PytorchStreamReader failed reading zip archive: failed finding central directory
```

**Comment by pharmapsychotic** [comment]
It required BLIP being installed as an "egg". Thanks to @justindujardin the BLIP install no longer needs to be egg. I've updated the instructions now. Let me know if this fixes things for ya!

```
# install torch with GPU support for example:
pip3 install torch torchvision --extra-index-url https://download.pytorch.org/whl/cu117

# install clip-interrogator and blip
pip install clip-interrogator==0.3.3
pip install git+https://github.com/pharmapsychotic/BLIP.git
```

**Comment by justindujardin** [comment]
Thanks for the excellent project and for absorbing the package changes. I can confirm that your install instructions work without my fork now.

BTW, If you wanted to simplify to just installing clip-interrogator, you could do it like this: https://github.com/justindujardin/clip-interrogator/commit/a766f19db6ee06f31f7a80d823224dfb3dad4682

**Comment by pharmapsychotic** [comment]
@justindujardin ooh finally a syntax that works with setup.py *BUT* ultimately didn't work with pip install 😢 `ERROR: Packages installed from PyPI cannot depend on packages which are not also hosted on PyPI.` 
So I think options are
a) post a BLIP package to pypi like ci-blip or something
b) just move the BLIP code to sub directory of this repo and bundle up together for easy pip install.

**Comment by ZetiMente** [comment]
@pharmapsychotic So there actually is already a BLIP package in pypi. 

https://pypi.org/project/blip-vit/

**Comment by pharmapsychotic** [comment]
@ZetiMente ooh nice find! I'll test that version out.

**Comment by ZetiMente** [comment]
> @ZetiMente ooh nice find! I'll test that version out.

Hmm, looking into it more, it might be too outdated. :/

**Comment by pharmapsychotic** [comment]
Okay! Thanks so much @justindujardin and @ZetiMente install should be simplified now to just `pip install clip-interrogator==0.3.5` \o/ 
@dante8300 can you give it a try? Everything work now for ya?

**Comment by dante8300** [comment]
Hi @pharmapsychotic, thanks for getting back. The error has changed slightly. Currently, I have 
torchvision --extra-index-url https://download.pytorch.org/whl/cu116 which matches the version I have but not sure if the diff in version is related.  Is there anything else i could try? 

```
Loading BLIP model...
Traceback (most recent call last):
  File "C:\lien\style-transfer\diff\clip-interrogator\run_cli.py", line 93, in <module>
    main()
  File "C:\ken\style-transfer\diff\clip-interrogator\run_cli.py", line 54, in main
    ci = Interrogator(config)
  File "C:\Users\ken\Anaconda2\envs\StableDiffusion\lib\site-packages\clip_interrogator\clip_interrogator.py", line 58, in __init__
    blip_model = blip_decoder(
  File "C:\Users\ken\Anaconda2\envs\StableDiffusion\lib\site-packages\blip\models\blip.py", line 175, in blip_decoder
    model,msg = load_checkpoint(model,pretrained)
  File "C:\Users\ken\Anaconda2\envs\StableDiffusion\lib\site-packages\blip\models\blip.py", line 218, in load_checkpoint
    checkpoint = torch.load(cached_file, map_location='cpu')
  File "C:\Users\ken\Anaconda2\envs\StableDiffusion\lib\site-packages\torch\serialization.py", line 777, in load
    with _open_zipfile_reader(opened_file) as opened_zipfile:
  File "C:\Users\ken\Anaconda2\envs\StableDiffusion\lib\site-packages\torch\serialization.py", line 282, in __init__
    super(_open_zipfile_reader, self).__init__(torch._C.PyTorchFileReader(name_or_buffer))
RuntimeError: PytorchStreamReader failed reading zip archive: failed finding central directory
```

**Comment by dante8300** [comment]
Seems to be stuck at loading the checkpoint. Not sure what to make to this error, any advice is welcome. 

This is how i am calling the client i guess nothing wrong there? `python run_cli.py -i "image.PNG"`

**Comment by alienfractal** [comment]
I found the parameters in the code, i think now its working.  One thing i noticed was that sometimes the downloading of the BLiP_MODEL halts almost at the end, which could be why it gets corrupted. But that was just a theory. 
![image](https://user-images.githubusercontent.com/8407437/211113982-1419aba7-4233-4498-b64d-0f3a72877356.png)
 
 I added the following parameters to the Interrogator Config, and it forced me to download the model again, updating the file in the process.  It works, but I see that 6GB in VRAM is not enough. 
 `from PIL import Image
from clip_interrogator import Config, Interrogator
image_path="sa1.PNG"
image = Image.open(image_path).convert('RGB')
ci = Interrogator(Config(clip_model_name="ViT-L-14/openai",blip_model_type="base",device="cuda",blip_max_length=24))
print(ci.interrogate(image))`
